### DISCUSSION TITLE
How to connect to RDS serverless Mysql using prisma when having a ssl certificate in lambda layer?

### DISCUSSION THREAD

**Discussion by MohamedAkl1** [prisma/prisma#20844] [Q&A]
### Question

I have an AWS Lambda layer that will have prisma in it's code to be used by all lambdas. The connection to the RDS must have an SSL certificate with it in order to connect to the RDS. I have a layer that is connected to the Lambda, In this layer i added Prisma schema and certificate in a Prisma folder as instructed in the docs. But whenever I'm trying to connect to the RDS in Lambda it always shows `Can't reach database server at...`I isolated everything else in a local environment to simulate the Lambda and I'm sure that the problem is with the SSL certificate.

I tried accessing the certificate from the lambda itself using the path `/opt/nodejs/prisma/certificate.pem` and it could be read and that is not a problem. My question is: Is there a specific way or a recommended one to access the certificate file or a best practice for Prisma when dealing with Lambda layers and ssl certificates in general?

### Expected behavior (optional)

Prisma should connect to RDS without a problem.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["rhel-openssl-1.0.x"]
}

datasource db {
  provider          = "mysql"
  url               = "mysql://prisma_db_user:${Password}@${HOST}:${PORT}/${Database}?sslcert=/opt/nodejs/rds-ca-2019-root.pem"
}

model post {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime
  title     String   @db.VarChar(255)
  content   String?
  published Boolean  @default(false)
  authorId  Int
  user      user     @relation(fields: [authorId], references: [id], map: "Post_authorId_fkey")

  @@index([authorId], map: "Post_authorId_fkey")
}

model profile {
  id     Int     @id @default(autoincrement())
  bio    String?
  userId Int     @unique(map: "Profile_userId_key")
  user   user    @relation(fields: [userId], references: [id], map: "Profile_userId_fkey")
}

model user {
  id      Int      @id @default(autoincrement())
  email   String   @unique(map: "User_email_key")
  name    String?
  post    post[]
  profile profile?
}

```

- Database: 
Prisma 5.1.1

**Comment by MohamedAkl1** [comment]
Commenting on this for better traction

**Comment by ludralph** [comment]
Hi @MohamedAkl1 👋 

Thank you for raising this question.

it seems that the issue you faced might be related to the path of the SSL certificate. As per this open [GitHub issue](https://github.com/prisma/prisma/issues/16768), the path to the SSL certificate can be different depending on the environment. On a Windows machine, the path is relative to the Prisma folder, but on a Linux environment (like AWS Lambda), it must be relative to root or perhaps elsewhere.

To address your specific question of whether there is a specific or recommended way to access the certificate file or a best practice for Prisma when dealing with Lambda layers and ssl certificates in general, I would say we don't have one documented yet and would want to create a document that shows how this can be achieved. Can you please open a [docs](https://github.com/prisma/docs/issues/new?assignees=&labels=docs&projects=&template=doc_request.md) issue so our team can look into documenting this.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nikolasburk** [reply]
@MohamedAkl1 were you able to resolve this in the end?

**Reply by owlyowl** [reply]
Has there been any further work on this out of interest I'd be keen to know if anyone has documented when using Lambdas and SSL

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
